app/api/ada/hardware-wallet/connectTrezorAdaWallet.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+// Handles Connecting Trezor with Yoroi that follow the v2 addressing scheme (bip44)
+
+import {
+  discoverAllAddressesFrom
+} from '../lib/utils';
+import {
+  saveAsAdaAddresses,
+  newAdaAddress
+} from '../adaAddress';
+import type {
+  AdaWallet,
+  AdaHardwareWalletParams
+} from '../adaTypes';
+import {
+  Logger,
+  stringifyError
+} from '../../../utils/logging';
+import { saveCryptoAccount, saveAdaWallet } from '../adaLocalStorage';
+import { createAdaHardwareWallet } from '../adaWallet';
+import { createHardwareWalletAccount } from '../adaAccount';
+import type { ConfigType } from '../../../../config/config-types';
+
+declare var CONFIG: ConfigType;
+const addressScanSize = CONFIG.app.addressScanSize;
+const addressRequestSize = CONFIG.app.addressRequestSize;
+
+/** Creates and stores(into local storage) a new/empty AdaWallet and CryptoAccount
+ * Fetches all related addresses
+ * Caches the fetched address results locally (into lovefieldDatabase)
+ * @returns a new AdaWallet
+ */
+export async function connectTrezorAdaWallet({
+  walletInitData
+}: AdaHardwareWalletParams): Promise<AdaWallet> {
+  try {
+    Logger.debug('connectTrezorAdaWallet::connectTrezorAdaWallet called');
+
+    // create ada wallet object for hardware wallet
+    const [adaWallet] = createAdaHardwareWallet({ walletInitData });
+    // create crypto account object for hardware wallet
+    // eslint-disable-next-line max-len
+    const cryptoAccount = createHardwareWalletAccount(walletInitData.cwHardwareInfo.publicMasterKey);
+
+    // fetch all External addresses
```

Comment:
everything from this line and below seems to be exactly the same as for the restore wallet code. Maybe it could be refactor and reused

Reply:
DONE